src/libs/ReportUtils.js

```diff
@@ -3043,6 +3043,27 @@ function shouldDisableRename(report, policy) {
     return !_.keys(loginList).includes(policy.owner) && policy.role !== CONST.POLICY.ROLE.ADMIN;
 }
 
+/**
+ * @param {String} policyID
+ * @param {String} policyName
+ * @returns {Object}
+ */
+function buildOptimisticExpenseChatForSchoolPrincipal(policyID, policyName) {
```

Comment:
I don't think we're building the optimistic chat for the school principal here, but rather the teacher. The principal's expense chat is created in the backend from the request initiated by the teacher. I think we can just reuse `buildOptimisticChatReport` in `AddSchoolPrincipal` action.